.github/workflows/release.yaml

```diff
@@ -0,0 +1,79 @@
+---
+name: Release
+
+run-name: release ${{ inputs.tag }}
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag to be released'
+        required: true
+
+permissions:
```

Comment:
giving write permissions to the whole workflow is somewhat of a security risk, best to limit the scope of these...

can we set the permissionson a single step, or at least a single job in the workflow (essetially the same here, but not the case in other workflows/repos)?

Reply:
We can't, it's either at the job level or workflow level.

Reply:
I'd put `packages: read` at the workflow level, and inside the `publish` job override and put `packages: write` if possible